source/SelectableGridComponent/SelectableGridComponent.js

```diff
@@ -0,0 +1,123 @@
+import { Component, PropTypes } from 'react'
+
+export default class SelectableGridComponent extends Component {
+
+  constructor (props) {
+    super(props)
+
+    this.state = {
+      selectedRows: props.selectedRows
+    }
+
+    this.handleCellStyle = this.handleCellStyle.bind(this)
+    this.handleOnSelect = this.handleOnSelect.bind(this)
+  }
+
+  static propTypes= {
+    /** Optional option to set if the a FlexTable allows multiple selection*/
+    allowsMultipleSelection: PropTypes.bool,
+
+    /** Children of the component, it only support the FlexTable so far*/
+    children: PropTypes.func.isRequired,
+
+    /**
+    * Optional callback when a row is selected
+    * ({ index: number }): void
+    */
+    onRowSelect: PropTypes.func,
+
+    /**
+    * Optional callback when a row is deselected
+    * ({ index: number }): void
+    */
+    onRowDeselect: PropTypes.func,
+
+    /**
+     * Callback invoked when a user clicks on a table row.
+     * ({ index: number }): void
+     */
+    onRowClick: PropTypes.func,
+
+    /** Optional custom CSS class for individual rows */
+    rowWrapperStyle: PropTypes.oneOfType([PropTypes.object, PropTypes.func]),
```

Comment:
Seems like users might want to react in more ways than just the `rowWrapperStyle`. Some users use external CSS (class names) instead of inline styles (eg `rowWrapperClassName`). Others may be using `rowClassName` or `rowStyle`. (The wrapper exists for kind of a specific purpose. I don't think it's the most commonly used one.)

Come to think of it, users might actually want to render different content (not just CSS changes).

It might be easier to satisfy all use-cases by allowing users to query `SelectableGridComponent` directly for its `selectedRows` value. Maybe something like...

```js
class MyComponent extends Component {
  render () {
    <SelectableGridComponent
      ref={(ref) => this._selectableGridComponent = ref}
      {...selectableGridComponentProps}
    >
      {({ onRowClick }) => (
        <FlexTable
          onRowClick={onRowClick}
          {...flexTableProps}
        >
          <FlexColumn
            {...flexColumnProps}
            rowClassName={({ index }) => {
              if (this._isRowSelected(index)) {
                return 'selectedRow'
              } else {
                return 'deselectedRow'
              }
            }}
            cellRenderer={
              ({ cellData, columnData, dataKey, rowData, rowIndex }) => {
                if (this._isRowSelected(index)) {
                  return <div>selected content</div>
                } else {
                  return <div>deselected content</div>
                }
              }
            }
          />
        </FlexTable>
      )}
    </SelectableGridComponent>
  }

  _isRowSelected (index) {
    return (
      this._selectableGridComponent &&
      this._selectableGridComponent.isRowSelected(index)
    )
  }
}
```